mmv1/products/vertexai/Model.yaml

```diff
@@ -0,0 +1,257 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: Model
+base_url: projects/{{project}}/locations/{{region}}/models
+create_url: projects/{{project}}/locations/{{region}}/models:upload
+self_link: 'projects/{{project}}/locations/{{region}}/models/{{name}}'
+update_verb: :PATCH
+update_mask: true
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Official Documentation':
+      'https://cloud.google.com/vertex-ai/docs'
+  api: 'https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.models'
+async: !ruby/object:Api::OpAsync
+  actions:
+    - create
+    - delete
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: True
+    allowed:
+      - True
+      - False
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+description: "A trained machine learning Model."
+id_format: 'projects/{{project}}/locations/{{region}}/models/{{name}}'
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "vertex_ai_model_basic"
+    primary_resource_id: "model"
+    vars:
+      name: "vertex_ai_model"
+      project: "vertex-ai"
+      model_name: "model-name"
+  - !ruby/object:Provider::Terraform::Examples
+    name: "vertex_ai_model_advanced"
+    primary_resource_id: "model"
+    vars:
+      name: "vertex_ai_model"
+      project: "vertex-ai"
+      model_name: "model-name"
+      kms_key_name: "kms-name"
+    test_vars_overrides:
+      kms_key_name: 'BootstrapKMSKeyInLocation(t, "us-central1").CryptoKey.Name'
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  decoder: templates/terraform/decoders/vertex_model.go.erb
+  encoder: templates/terraform/encoders/vertex_model.go.erb
+  update_encoder: templates/terraform/encoders/vertex_model_update.go.erb
+parameters:
+  - !ruby/object:Api::Type::String
+    name: region
+    description: The location for the resource
+    url_param_only: true
+    required: true
+    immutable: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: name
+    description: The resource name of the Model.
+    required: true
+    immutable: true
+    custom_flatten: templates/terraform/custom_flatten/name_from_self_link.erb
+  - !ruby/object:Api::Type::String
+    name: versionId
+    description: Output only. Immutable. The version ID of the model. A new version is committed when a new model version is uploaded or trained under an existing model id. It is an auto-incrementing decimal number in string representation.
+    output: true
+  - !ruby/object:Api::Type::Array
+    name: versionAliases
```

Comment:
I think the version UX is fairly undefined- how would a user interact with multiple versions using Terraform?

Reply:
Terraform will only interact with the first version of a model.